raphtory-graphql/src/model/mod.rs

```diff
@@ -124,24 +125,39 @@ pub(crate) struct Mut(MutRoot);
 
 #[MutationFields]
 impl Mut {
-    async fn rename_graph<'a>(
+    // If namespace is not provided, it will be set to the current working directory.
+    async fn delete_graph<'a>(
         ctx: &Context<'a>,
-        parent_graph_name: String,
-        parent_graph_namespace: &Option<String>,
         graph_name: String,
         graph_namespace: &Option<String>,
-        new_graph_name: String,
     ) -> Result<bool> {
         let data = ctx.data_unchecked::<Data>();
-        let parent_graph_path =
-            construct_graph_path(&data.work_dir, &parent_graph_name, parent_graph_namespace)?;
-        if !parent_graph_path.exists() {
+        let current_graph_path =
+            construct_graph_path(&data.work_dir, &graph_name, graph_namespace)?;
+        if !current_graph_path.exists() {
             return Err(GraphError::GraphNotFound(construct_graph_name(
-                &parent_graph_name,
-                parent_graph_namespace,
+                &graph_name,
+                graph_namespace,
             ))
             .into());
         }
+
+        delete_graph(&current_graph_path)?;
+        data.graphs.remove(&graph_name);
+
+        Ok(true)
+    }
+
+    // If namespace is not provided, it will be set to the current working directory.
+    // This applies to both the graph namespace and new graph namespace.
+    async fn move_graph<'a>(
```

Comment:
Feels like move could just call `copy` then `delete`? Means we would only need to change the functions in one place